packages/@sfcc-bff/cartapi/src/api/schema/cartResolvers.js

```diff
@@ -0,0 +1,189 @@
+import CommerceSdk from 'commerce-sdk';
+
+const { ApolloError } = apollo;
+
+const getBasketClient = async (config, context) => {
+    const clientConfig = getCommerceClientConfig(config);
+    clientConfig.headers.authorization = (
+        await getUserFromContext(context)
+    ).token;
+    return new CommerceSdk.Checkout.ShopperBaskets(clientConfig);
+};
+
+const addProductToBasket = async (productId, quantity, config, context) => {
+    const basketClient = await getBasketClient(config, context);
+    let basketId = context.getSessionProperty('basketId');
+    if (!basketId) {
+        let customerBasket = await basketClient.createBasket({
+            body: {},
+        });
+        context.setSessionProperty('basketId', customerBasket.basketId);
+        basketId = customerBasket.basketId;
+    }
+
+    return basketClient.addItemToBasket({
+        parameters: {
+            basketId: basketId,
+        },
+        body: [{ productId: productId, quantity: quantity }],
+    });
+};
+
+const getBasket = async (config, context) => {
+    let basketId = context.getSessionProperty('basketId');
+    // If No Cart has been created yet, return error
+    if (!basketId) {
+        return {
+            fault: {
+                type: 'NoCartCreated',
+                message: 'No Cart has been created yet.',
+            },
+        };
+    }
+    // else get basket with that id
+    const basketClient = await getBasketClient(config, context);
+
+    let basket = await basketClient.getBasket({
+        parameters: {
+            basketId: basketId,
+        },
+    });
+
+    if (basket.fault) {
+        return basket;
+    }
+
+    basket.getCartMessage = `Basket found with ID of ${basketId}`;
+
+    // Get Shipping Methods
+    const shipmentId = basket.shipments[0].shipmentId;
+    const shippingMethods = await getShippingMethods(
+        basketId,
+        shipmentId,
+        config,
+        context,
+    );
+    if (shippingMethods.fault) {
+        return shippingMethods;
+    }
+    // Update Shipping Method to Default Method
+    let selectedShippingMethodId = basket.shipments[0].shippingMethod
+        ? basket.shipments[0].shippingMethod.id
+        : shippingMethods.defaultShippingMethodId;
+
+    basket = await updateShippingMethod(
+        shipmentId,
+        selectedShippingMethodId,
+        config,
+        context,
+    );
+
+    basket.shippingMethods = shippingMethods;
+    return basket;
+};
+
+const getShippingMethods = async (basketId, shipmentId, config, context) => {
+    const basketClient = await getBasketClient(config, context);
+
+    return basketClient.getShippingMethodsForShipment({
+        parameters: {
+            basketId: basketId,
+            shipmentId: shipmentId,
+        },
+    });
+};
+
+const updateShippingMethod = async (
+    shipmentId,
+    shippingMethodId,
+    config,
+    context,
+) => {
+    const basketId = context.getSessionProperty('basketId');
```

Comment:
Why don't we pass in basketId like we do in getShippingMethods?

Reply:
`updateShippingMethod` can be called by the external (UI) components. `getShippingMethods` is only called within `cartResolvers.js`.

Reply:
Great to know! Thanks